logstore: rm old sideloaded storage migration

pav-kv opened the pull request:
This change removes the sideloaded storage path migration introduced in #35035. It has been safe to remove this migration since v20.1.

This reduces the cost of a replica start-up, because it no longer needs to call into the file system when creating the sideloaded storage.

Fixes #94811
Epic: none
Release note: none

pav-kv committed:
logstore: rm old sideloaded storage migration

This change removes the sideloaded storage path migration introduced in #35035.
It has been safe to remove this migration since v20.1.

This reduces the cost of a replica start-up, because it no longer needs to call
into the file system when creating the sideloaded storage.

Fixes #94811
Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94828)
<!-- Reviewable:end -->

pav-kv committed:
logstore: tidy up NewDiskSideloadStorage

Remove the unused error from the returned values, inline some bits.

Epic: none
Release note: none

tbg reviewed (commented):
Thank you!

pavelkalinnikov commented:
bors r=tbg
